checks.d/linux_proc_extras.py

```diff
@@ -0,0 +1,87 @@
+# project
+from checks import AgentCheck
+import utils.subprocess_output
+import subprocess
+
+PROCESS_STATES = {
+    'D': 'uninterruptible',
+    'R': 'runnable',
+    'S': 'sleeping',
+    'T': 'stopped',
+    'W': 'paging',
+    'X': 'dead',
+    'Z': 'zombie',
+}
+
+PROCESS_PRIOS = {
+    '<': 'high',
+    'N': 'low',
+    'L': 'locked'
+}
+
+class MoreUnixCheck(AgentCheck):
+    def check(self, instance):
+        tags = instance.get('tags', [])
+
+        state_counts = {
+            'uninterruptible': 0,
+            'runnable': 0,
+            'sleeping': 0,
+            'stopped': 0,
+            'paging': 0,
+            'dead': 0,
+            'zombie': 0
+        }
+
+        prio_counts = {
+            'low': 0,
+            'high': 0,
+            'locked': 0
+        }
+
+        try:
+            with open('/proc/sys/fs/inode-nr', 'r') as inode_info:
+                inode_stats = inode_info.readline().split()
+                self.gauge('system.inodes.total', float(inode_stats[0]), tags=tags)
+                self.gauge('system.inodes.used', float(inode_stats[1]), tags=tags)
+
+            with open('/proc/stat', 'r') as stat_info:
+                lines = [line.strip() for line in stat_info.readlines()]
+
+                for line in lines:
+                    if line.startswith('ctxt'):
+                        ctxt_count = float(line.split(' ')[1])
+                        self.monotonic_count('linux.context_switches', ctxt_count, tags=tags)
```

Comment:
I chose `linux` because I had no intention of dealing with other Unixes. Would you like me to do `system.linux.*` or just `system.*` here?